plugins/trace_api_plugin/include/eosio/trace_api/chain_extraction.hpp

```diff
@@ -100,11 +102,15 @@ class chain_extraction_impl_type {
             if( it != cached_traces.end() ) {
                traces.emplace_back( to_transaction_trace<transaction_trace_t>( it->second ));
             }
+            tt.ids.emplace_back(id);
          }
          clear_caches();
 
-         store.append( std::move( bt ) );
+         // tt entry acts as a placeholder in a trx id slice if this block has no transaction
+         tt.block_num = bt.number;
+         store.append_trx_ids( tt );
```

Comment:
Should add a `std::move( tt )`

Reply:
> This could be taken by value and then moved into the metadata_log_entry to avoid an extra copy.

I removed `std::move(tt)` in commit 09ae42a while working on the comments above. Should I add it back?

Reply:
Done.